src/metabase/models/native_query_snippet.clj

```diff
@@ -84,25 +85,31 @@
   [_model-name _opts snippet]
   (-> (serdes/extract-one-basics "NativeQuerySnippet" snippet)
       (update :creator_id serdes/*export-user*)
-      (update :collection_id #(when % (serdes/*export-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*export-fk* % 'Collection))))
 
 (defmethod serdes/load-xform "NativeQuerySnippet" [snippet]
   (-> snippet
       serdes/load-xform-basics
       (update :creator_id serdes/*import-user*)
-      (update :collection_id #(when % (serdes/*import-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*import-fk* % 'Collection))))
 
 (defmethod serdes/dependencies "NativeQuerySnippet"
   [{:keys [collection_id]}]
   (if collection_id
     [[{:model "Collection" :id collection_id}]]
     []))
 
-(defmethod serdes/storage-path "NativeQuerySnippet" [snippet ctx]
-  ;; Intended path here is ["snippets" "nested" "collections" "snippet_eid_and_slug"]
-  ;; We just the default path, then pull it apart.
-  ;; The default is ["collections" "nested" collections" "nativequerysnippets" "base_name"]
-  (let [basis  (serdes/storage-default-collection-path snippet ctx)
-        file   (last basis)
-        colls  (->> basis rest (drop-last 2))] ; Drops the "collections" at the start, and the last two.
-    (concat ["snippets"] colls [file])))
+(defmethod serdes/storage-path "NativeQuerySnippet" [snippet _ctx]
+  (let [{:keys [id label]} (-> snippet serdes/path last)]
+    ["snippets" (serdes/storage-leaf-file-name id label)]))
+
+(defmethod serdes/load-one! "NativeQuerySnippet" [ingested maybe-local]
+  (if (and (not (and maybe-local
+                     (= (:name ingested) (:name maybe-local))))
+           (t2/select-one :model/NativeQuerySnippet
+                          :name (:name ingested) :entity_id [:!= (:entity_id ingested)]))
+    (recur (update ingested :name
+                   #(str % "-" (or (:id maybe-local)
+                                   (subs (:entity_id ingested) (- (count (:entity_id ingested)) 4)))))
```

Comment:
```suggestion
                   #(str % "- 2"))))
```
I don't think it makes sense to have the ingested snippet postfix with the id of the local snippet.

We can post-fix it with `- 2` because we can't have multiple snippets with the same name anw.

Reply:
That would be a rare situation, but you could have loaded another serialization with a snippet with the same name and it's already in db having -2 in its name.

Reply:
in that case, this will update the new name to have postfix twice right? sth like "some snippet -2 -2"

Reply:
🤦  right! So I guess using ` (copy)` is fine then 😁

Reply:
done